server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -172,9 +198,62 @@ protected void startRunningCleaner(ScheduledThreadPoolExecutor schedExecutor) {
     ThreadPools.watchNonCriticalScheduledTask(future);
   }
 
+  private void processRefreshes(Ample.DataLevel dataLevel) {
+
+    // process batches of refresh entries to avoid reading all into memory at once
+    Iterators.partition(ctx.getAmple().refreshes(dataLevel).list().iterator(), 10000)
+        .forEachRemaining(refreshEntries -> {
+          LOG.info("Processing {} tablet refreshes for {}", refreshEntries.size(), dataLevel);
+
+          var extents =
+              refreshEntries.stream().map(RefreshEntry::getExtent).collect(Collectors.toList());
+          var tabletsMeta = new HashMap<KeyExtent,TabletMetadata>();
+          ctx.getAmple().readTablets().forTablets(extents, Optional.empty())
+              .fetch(PREV_ROW, LOCATION, SCANS).build().stream()
```

Comment:
I got a warning from my IDE that the `TabletsMetadata` object created by `.build()` is not closed here.

Reply:
Fixed in dedaa88